### PR TITLE
Tests/use fake timers

### PR DESCRIPTION
**Pull request by rluvaton**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: test/route-rate-limit.test.js [hunk 1]
```diff
@@ -226,7 +239,7 @@ test('With redis store', async t => {
     message: 'Rate limit exceeded, retry in 1 second'
   }, JSON.parse(res.payload))
 
-  // TODO - use sinom timers
+  // Not using fake timers here as we use an external Redis which wouldn't be effected by this
```
Comment (Fdawgs) [inline test/route-rate-limit.test.js:242]:
```suggestion
  // Not using fake timers here as we use an external Redis that would not be effected by this
```

File: test/global-rate-limit.test.js [hunk 2]
```diff
@@ -271,7 +279,7 @@ test('With redis store', async t => {
     message: 'Rate limit exceeded, retry in 1 second'
   }, JSON.parse(res.payload))
 
-  // TODO - use sinom timers
+  // Not using fake timers here as we use an external Redis which wouldn't be effected by this
```
Comment (Fdawgs) [inline test/global-rate-limit.test.js:282]:
```suggestion
  // Not using fake timers here as we use an external Redis that would not be effected by this
```

File: store/LocalStore.js [hunk 3]
```diff
@@ -29,7 +29,7 @@ LocalStore.prototype.incr = function (ip, cb) {
     cb(null, { current, ttl: this.timeWindow })
   } else {
     // start counting from the first request/increment
-    if (!this.msLastBeat) {
+    if (this.msLastBeat === undefined || this.msLastBeat === null) {
```
Comment (Fdawgs) [inline store/LocalStore.js:32]: The existing logical not operator already catches falsy values, mind clarifying on this change?

Comment (rluvaton) [reply store/LocalStore.js:32]: Yes, of course

Sinon fake timers set the starting time to be 0 and it's false so...

Comment (rluvaton) [reply store/LocalStore.js:32]: I know but:
1. this helps with debugging as we can see how much time passed
2. #217 remove this check anyway...

File: test/route-rate-limit.test.js (hunk 1 above)
Comment (rluvaton) [reply test/route-rate-limit.test.js]: Thanks!

File: test/global-rate-limit.test.js (hunk 2 above)
Comment (rluvaton) [reply test/global-rate-limit.test.js]: Thanks!

File: store/LocalStore.js (hunk 3 above)
Comment (mcollina) [reply store/LocalStore.js:32]: Ok!

**Review by mcollina — APPROVED** [review]
lgtm

**Review by Fdawgs — APPROVED** [review]

**Comment by rluvaton** [comment]
If everyone accept it I believe it can be merged 😀

**Pull request merged**